src/integrations/Elastic.Apm.AspNetFullFramework/Extensions/SoapRequest.cs

```diff
@@ -20,6 +20,8 @@ internal static class SoapRequest
 		private const string SoapActionHeaderName = "SOAPAction";
 		private const string ContentTypeHeaderName = "Content-Type";
 		private const string SoapAction12ContentType = "application/soap+xml";
+		private const int MaxRequestBodySizeInBytesToCapture = 1024 * 10;
+		private static readonly ArrayPool<byte> ArrayPool = ArrayPool<byte>.Create(MaxRequestBodySizeInBytesToCapture, 50);
```

Comment:
`maxArraysPerBucket = 50` is [the default](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Private.CoreLib/src/System/Buffers/ConfigurableArrayPool.cs#L15C63-L15C65).